docs/service-and-repository-structure/service-and-repository-structure.md

```diff
@@ -0,0 +1,137 @@
+RPC connection.  
+The `Headless Server` wraps the client action to blockchain Transaction(Tx) and propagate to remote RPC server to be
+mined inside new block.  
+`Headless Server` also runs GraphQL server at the same time, so the [`Launcher`](#2.1.-Launcher) can query and get the
+newest data.
+
+### 2.2.1. Remark
+
+`Headless Server` is embedded inside [`Launcher`](#2.1.-Launcher) at this moment for development convenience and
+historical issue, and it will be moved out from the [`Launcher`](#2.1.-Launcher).
+
+## 2.3. Unity Client
+
+`Unity Client` is what you play actual NineChronicles game. All user interactions occurs in this player.  
+`Unity Client` is connected to [`Headless Server`](#2.2.-Headless-Server) using RPC protocol to send/receive data from
+blockchain and render new states to the screen.
+
+# 3. Repository and Library Structure
+
+![NineChronicles Repository Structure](images/0002_repository_structure.png)
+
+Each part of NineChronicles has common libraries to communicate each other.  
+In this chapter, We'll check the core libraries to run NineChronicles.  
+After read this chapter, you would know where to see and touch to modify NineChronicles as you wish.
+
+## 3.1. 9c-launcher
+
+#### [GitHub repository](https://github.com/planetarium/9c-launcher)
+
+The repository for [`Launcher`](#2.1.-Launcher). You can build and run your own NineChronicles launcher using this repository.  
+`9c-launcher` is mostly written in TypeScript and you can easily build and run launcher following [Getting Started](https://github.com/planetarium/9c-launcher/wiki/Getting-Started).  
+If you build and run your launcher with default setting, the launcher will attach to NineChronicles mainnet(the production server).  
+Otherwise you can attach to different(maybe your own) blockchain network, you can achieve that through editing `config.json` to change target network.
+
+## 3.2. NineChronicles.Headless
+
+#### [GitHub repository](https://github.com/planetarium/NineChronicles.Headless)
+
+The repository for [`Headless Server`](#2.2.-Headless-Server). You can run your own blockchain node whether it is full node or not.  
+`NineChronicles.Headless` is mostly written in C#. You can build and run your local headless server
+following [Getting Started](https://github.com/planetarium/NineChronicles.Headless/wiki/Getting-Started).
+You can create and run your own blockchain network/node with this repository. Please
+check [Create new genesis block](https://github.com/planetarium/NineChronicles.Headless/wiki/Create-new-genesis-block)
+to create your own genesis block for your own network.
+
+### 3.2.1. Remark
+#### 3.2.1.1. Options for headless node
+There are several functions can be run using this repository.
+
+- Peer: save & spread available node addresses.
+- RPC server: Receives action data from client and make transaction to spread to peers.
+- GraphQL server: Returns data from blockchain to answer incoming GQL query.
```

Comment:
GraphQL server also provides the node's status and APIs to create actions. But I'm not sure that we can recommend `actionQuery` 🤔 So at this time...
```suggestion
- GraphQL server: Provides blockchain data and the node's status to answer incoming GQL query.
```

Reply:
> GraphQL server also provides the node's status and APIs to create actions. But I'm not sure that we can recommend `actionQuery` thinking So at this time...

I also concern about `mutation` type query. But that is there and we should not hide those things because it has possibility of abusing.
I didn't mentioned at my first version because I forgot :rofl: